sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/HL7v2IO.java

```diff
@@ -466,21 +475,63 @@ private Message fetchMessage(HealthcareApiClient client, String msgId)
       this.hl7v2Stores = hl7v2Stores;
       this.filter = filter;
       this.initialSplitDuration = null;
+      this.timestampMethod = DEFAULT_TIMESTAMP_METHOD;
     }
 
-    public ListHL7v2Messages withInitialSplitDuration(Duration initialSplitDuration) {
+    /**
+     * Controls the initial splitting for parallelization of HL7v2 Messages List requests based on
+     * disjoint sendTime filters of the specified duration.
+     *
+     * @param initialSplitDuration the initial split duration
+     * @return the list hl 7 v 2 messages
+     */
+    ListHL7v2Messages withInitialSplitDuration(Duration initialSplitDuration) {
       this.initialSplitDuration = initialSplitDuration;
```

Comment:
Please return a new version of this transform instead of mutating the existing version.

It is not uncommon for users to take one transform apply it to the graph once and then call withX on it and apply it elsewhere. We wouldn't want the second mutation to ever impact the first application of the transform and it is easy to guard against it using the @AutoValue builders.